src/smt/theory_str_noodler/var_union_find.h

```diff
@@ -0,0 +1,79 @@
+#ifndef _VAR_UNION_FIND_
+#define _VAR_UNION_FIND_
+
+#include <functional>
+#include <list>
+#include <set>
+#include <stack>
+#include <map>
+#include <memory>
+#include <queue>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include <smt/params/smt_params.h>
+#include "ast/arith_decl_plugin.h"
+#include "ast/seq_decl_plugin.h"
+#include "smt/params/theory_str_noodler_params.h"
+#include "smt/smt_kernel.h"
+#include "smt/smt_theory.h"
+#include "smt/smt_arith_value.h"
+#include "util/scoped_vector.h"
+#include "util/union_find.h"
+#include "ast/rewriter/seq_rewriter.h"
+#include "ast/rewriter/th_rewriter.h"
+
+namespace smt::noodler {
+
+    typedef std::vector<std::set<BasicTerm>> BasicTermEqiv;
+
+    static bool ec_are_equal(const BasicTermEqiv& ec, const BasicTerm& t1, const BasicTerm& t2) {
+        for(const auto& st: ec) {
+            if(st.find(t1) != st.end() && st.find(t2) != st.end()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    class var_union_find {
```

Comment:
Some comments about what this class is for would be nice

Reply:
I added some comments